editoast/src/client/garbage_collector.rs

```diff
@@ -0,0 +1,104 @@
+use crate::models::timetable::Timetable;
+use database::DbConnectionPoolV2;
+use database::tables::scenario::dsl as scenario_dsl;
+use database::tables::stdcm_search_environment::dsl as stdcm_dsl;
+use database::tables::timetable::dsl as timetable_dsl;
+use diesel::prelude::*;
+use diesel_async::RunQueryDsl;
+use diesel_async::scoped_futures::ScopedFutureExt;
+use editoast_models::prelude::*;
+use std::ops::DerefMut;
+use std::sync::Arc;
+
+pub async fn run_garbage_collector(db_pool: Arc<DbConnectionPoolV2>) -> anyhow::Result<()> {
+    clean_orphaned_timetables(&db_pool).await?;
+    Ok(())
+}
+
+/// Deletes timetables that are not referenced by any Scenario or StdcmSearchEnvironment
+async fn clean_orphaned_timetables(db_pool: &Arc<DbConnectionPoolV2>) -> anyhow::Result<()> {
+    let conn = &mut db_pool.get().await?;
+
+    conn.clone()
+        .transaction(|conn| {
+            async move {
+                let orphaned_ids: Vec<i64> = timetable_dsl::timetable
+                    .left_join(
+                        scenario_dsl::scenario.on(scenario_dsl::timetable_id.eq(timetable_dsl::id)),
+                    )
+                    .left_join(
+                        stdcm_dsl::stdcm_search_environment
+                            .on(stdcm_dsl::timetable_id.eq(timetable_dsl::id)),
+                    )
+                    .filter(scenario_dsl::id.is_null())
+                    .filter(stdcm_dsl::id.is_null())
+                    .select(timetable_dsl::id)
+                    .load::<i64>(conn.write().await.deref_mut())
+                    .await?;
+
+                if orphaned_ids.is_empty() {
+                    println!("No orphaned timetables found");
+                    return Ok(());
+                }
+
+                let deleted_count =
+                    Timetable::delete_batch(&mut conn.clone(), orphaned_ids).await?;
+                println!("✅ {} orphaned timetable(s) deleted", deleted_count);
+
+                Ok(())
+            }
+            .scope_boxed()
+        })
+        .await
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::models::fixtures::{create_scenario_fixtures_set, create_timetable};
```

Comment:
nitpick: we try to flatten the `use`.


```suggestion
    use crate::models::fixtures::create_scenario_fixtures_set;
    use crate::models::fixtures::create_timetable;
```

Reply:
beat me to it ^^

Reply:
Done